ee/clickhouse/queries/funnels/funnel_unordered.py

```diff
@@ -0,0 +1,74 @@
+from typing import List
+
+from ee.clickhouse.queries.funnels.base import ClickhouseFunnelBase
+
+
+class ClickhouseFunnelUnordered(ClickhouseFunnelBase):
+    """
+    Unordered Funnel is a funnel where the order of steps doesn't matter.
+
+    ## Query Intuition
+
+    Imagine a funnel with three events: A, B, and C.
+    This query splits the problem into two parts:
+    1. Given the first event is A, find the furthest everyone went starting from A.
+       This finds any B's and C's that happen after A (without ordering them)
+    2. Repeat the above, assuming first event to be B, and then C.
+    
+    Then, the outer query unions the result of (2) and takes the maximum of these.
+
+    ## Results
+
+    The result format is the same as the basic funnel, i.e. [step, count].
+    Here, `step_i` (0 indexed) signifies the number of people that did at least `i+1` steps.
+    """
+
+    def get_query(self, format_properties):
+        return self.get_step_counts_query()
+
+    def get_step_counts_query(self):
+
+        max_steps = len(self._filter.entities)
+        union_queries = []
+        entities_to_use = list(self._filter.entities)
+
+        partition_select = self.get_partition_cols(1, max_steps)
+        sorting_condition = self.get_sorting_condition(max_steps)
+
+        for i in range(max_steps):
```

Comment:
BTW what is our convention for numbering steps in queries? I'm not actually sure, but I saw that https://github.com/PostHog/posthog/pull/4892 used 1-based indexing, while this is 0-based

Reply:
step_i is zero based everywhere, while level_index is one based everywhere :/ - idea is to get everything to zero based, soonish. I wrote a comment mentioning this on the first Funnel PR - https://github.com/PostHog/posthog/pull/4851#discussion_r657187802

Reply:
Please feel free to merge if you're happy with it :) (comment if not lol)

Reply:
Ah, I see now, I just misread #4892 a bit :+1: